scripts/setup/inline_email_css.py

```diff
@@ -27,6 +27,40 @@ def configure_cssutils() -> None:
     profile.addProfiles([(Profiles.CSS_LEVEL_2, properties[Profiles.CSS_LEVEL_2],
                          macros[Profiles.CSS_LEVEL_2])])
 
+def inline_template(template_name: str) -> None:
+    with open(os.path.join(EMAIL_TEMPLATES_PATH, template_name)) as template_source_file:
+        template_str = template_source_file.read()
+
+    output = Premailer(template_str,
+                       external_styles=[os.path.join(EMAIL_TEMPLATES_PATH, "email.css")]).transform()
+    output = escape_jinja2_characters(output)
+
+    # Premailer.transform will try to complete the DOM tree,
+    # adding html, head, and body tags if they aren't there.
+    # While this is correct for the email_base_default template,
+    # it is wrong for the other templates that extend this
+    # template, since we'll end up with 2 copies of those tags.
+    # Thus, we strip this stuff out if the template extends
+    # another template.
+    # email_base_default does not have this tags, by design
+    if template_name != 'email_base_default.source.html':
+        output = strip_unnecesary_tags(output)
+
+    if ('zerver/emails/compiled/email_base_default.html' in output or
+            'zerver/emails/email_base_messages.html' in output):
+        assert output.count('<html>') == 0
+        assert output.count('<body>') == 0
+        assert output.count('</html>') == 0
+        assert output.count('</body>') == 0
+
+    compiled_template_path = get_compiled_template_path(template_name)
+    os.makedirs(COMPILED_EMAIL_TEMPLATES_PATH, exist_ok=True)
+    with open(compiled_template_path, 'w') as final_template_file:
+        final_template_file.write(output)
+
+def get_compiled_template_path(template_name: str) -> str:
+    return os.path.join(COMPILED_EMAIL_TEMPLATES_PATH, template_name.split('source.html')[0] + "html")
```

Comment:
This commit makes some random extra changes during the extraction:

![image](https://user-images.githubusercontent.com/2746074/79027432-ffebfa80-7b40-11ea-8e8e-7072d31fee91.png)

(the `git show` `color-moved` option is really useful for seeing this sort of thing).

These changes fix the bug I noted in the previous commit, which is why the end result of the branch works.